/* protoc mesg.proto --python_out=. */

package mthyme;

message ExecuteCode {
    required string code = 1;  
    optional bool preparse = 2 [default = true];
}

message StartSession {
    optional int32 max_walltime = 1;
    optional int32 max_cputime = 2;
    optional int32 max_numfiles = 3;
    optional int32 max_vmem = 4;
}

message TerminateSession {
}

message SessionDescription {
    required int32 pid = 1;
}

message SendSignal {
    required int32 pid = 1;
    required int32 signal = 2;    
}

message Output {
    optional string stdout = 1;
    optional string stderr = 2;
    optional bool done = 3 [default = false];
}

message Message {
    enum Type { EXECUTE_CODE = 1; START_SESSION = 2; TERMINATE_SESSION = 3;
                SESSION_DESCRIPTION = 4; SEND_SIGNAL = 5;  OUTPUT = 6;}

    required Type   type = 1;
    optional uint32 id = 2;

    optional ExecuteCode execute_code = 3;
    optional StartSession start_session = 4;
    optional StartSession terminate_session = 5;
    optional SessionDescription session_description = 6;
    optional SendSignal send_signal = 7;
    optional Output output = 8;
}



